atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServices.java

```diff
@@ -88,19 +88,18 @@ static void waitForSchemaVersions(
         long sleepTime = INITIAL_SLEEP_TIME;
         Map<String, List<String>> versions;
         do {
-            // This only returns the schema versions of nodes that the client knows exist. In particular, if a node we
-            // shook hands with goes down, it will have schema version UNREACHABLE; however, if we never shook hands
-            // with a node, there will simply be no entry for it in the map. Hence the check for the number of nodes.
+            // This may only include some of the nodes if the coordinator hasn't shaken hands with someone; however,
```

Comment:
As far as I know this state is impossible to be in provided you have a quorum nodes, and therefore I don't believe is a legitimate concern.

The primary goal of this method in my opinion (and we should directly document this in the javadoc is):
- backoff doing schema mutations if the cluster is known to be in disagreement currently
  - I think this is just sensible and a natural thing to do, but admit given recent (read: when we did all the rescue restore optimization work) schema optimization work in our fork that this is less of a concern
- don't do a schema mutation if it can potentially result in "split-brain" (from the schema's point of view) in the future
  - The code as written in this PR allows for the case that we 1 out of 3 nodes are online, we push a schema mutation to that node. That node goes down, two other nodes come online, a schema mutation is pushed to those two nodes (whether by us or by another atlasdb client). The original node comes online, the nodes are in disagreement. While cassandra in theory should handle this case fine, the more divergent the schemas get the harder it will be for it to rectify them. Additionally there is the nuance/complexity around each different version of the schema having matching new pieces (like same keyspace/table name), but with a different cf_id. While I believe we've resolved this via deterministic cf_ids its worth pointing out that for that case, we would treat the situation as an outage and require manual intervention. We can prevent all the risks of the "split-brain" case here, by ensuring we only apply changes when we believe a majority of nodes are going to receive. From a maintain "HA" perspective this should be fine: see my next bullet.
- schema mutations are HA
  - We want to make sure critical path operations in AtlasDb are HA. We believe modifying a schema to be one of those things and thus it should be HA. That being said the rest of the critical path operations in AtlasDb, even when HA, have limitations to their availability. The most obvious of which are basic reads/writes to cassandra: they require quorum. Given the way we deploy cassandra (network topology aware RF=3), at any given moment [in the best case] if we have more than a third of the nodes of the cluster offline we have lost availability and can no longer read/write. Because of this I do not view being able to do a schema mutation where less than 2/3 of the nodes of the cluster are online as necessary.

Reply:
Given my statements above, even if there was a fear that describe_schema_versions() might not return a node that is freshly bootstrapping/about-to-bootstrap (which we only have at most 1 of at a time), we could set the requirement to simple-majority i.e. `floor(N/2)+1` rather than quorum (which is actually hard to accurately calculate in terms of raw numbers but is a stricter requirement to meet than simple-majority).

Reply:
Yep, I'll change this to require a simple-majority. I'll push that first, and then look at the docs/reorganisation of the code here.

Reply:
(Added the docs now.)